diff -Naur gdal-3.2.1-old/frmts/raw/envidataset.cpp gdal-3.2.1-new/frmts/raw/envidataset.cpp
--- gdal-3.2.1-old/frmts/raw/envidataset.cpp	2023-06-02 14:29:21.000000000 -0700
+++ gdal-3.2.1-new/frmts/raw/envidataset.cpp	2023-06-02 14:33:55.000000000 -0700
@@ -1954,12 +1954,12 @@
 bool ENVIDataset::ReadHeader( VSILFILE *fpHdr )
 
 {
-    CPLReadLine2L(fpHdr, 10000, nullptr);
+    CPLReadLine2L(fpHdr, -1, nullptr);
 
     // Start forming sets of name/value pairs.
     while( true )
     {
-        const char *pszNewLine = CPLReadLine2L(fpHdr, 10000, nullptr);
+        const char *pszNewLine = CPLReadLine2L(fpHdr, -1, nullptr);
         if( pszNewLine == nullptr )
             break;
 
@@ -1973,7 +1973,7 @@
             osWorkingLine.find("}") == std::string::npos )
         {
             do {
-                pszNewLine = CPLReadLine2L(fpHdr, 10000, nullptr);
+                pszNewLine = CPLReadLine2L(fpHdr, -1, nullptr);
                 if( pszNewLine )
                 {
                     osWorkingLine += pszNewLine;
